doc/hardware/peripherals/can/shell.rst

```diff
@@ -0,0 +1,266 @@
+.. _can_shell:
+
+CAN Shell
+#########
+
+.. contents::
+    :local:
+    :depth: 1
+
+Overview
+********
+
+The CAN shell provides a ``can`` command with a set of subcommands for the :ref:`shell <shell_api>`
+module. It allows for testing and exploring the :ref:`can_api` driver API through an interactive
+interface without having to write a dedicated application. The CAN shell can also be enabled in
+existing applications to aid in interactive debugging of CAN issues.
+
+The CAN shell provides access to most CAN controller features, including inspection, configuration,
+sending and receiving of CAN frames, and bus recovery.
+
+In order to enable the CAN shell, the following :ref:`Kconfig <kconfig>` options must be enabled:
+
+* :kconfig:option:`CONFIG_SHELL`
+* :kconfig:option:`CONFIG_CAN`
+* :kconfig:option:`CONFIG_CAN_SHELL`
+
+The following :ref:`Kconfig <kconfig>` options enable additional subcommands and features of the
+``can`` command:
+
+* :kconfig:option:`CONFIG_CAN_FD_MODE` enables CAN FD specific subcommands (e.g. for setting the
+  timing for the CAN FD data phase).
+* :kconfig:option:`CONFIG_CAN_RX_TIMESTAMP` enables printing of timestamps for received CAN frames.
+* :kconfig:option:`CONFIG_CAN_STATS` enables printing of various statistics for the CAN controller
+  in the ``can show`` subcommand. This depends on :kconfig:option:`CONFIG_STATS` being enabled as
+  well.
+* :kconfig:option:`CONFIG_CAN_AUTO_BUS_OFF_RECOVERY` enables the ``can recover`` subcommand when
+  disabled.
+
+For example, building the :ref:`hello_world` sample for the :ref:`frdm_k64f` with the CAN shell and
+CAN statistics enabled:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/hello_world
+   :board: frdm_k64f
+   :gen-args: -DCONFIG_SHELL=y -DCONFIG_CAN=y -DCONFIG_CAN_SHELL=y -DCONFIG_STATS=y -DCONFIG_CAN_STATS=y
+   :goals: build
+
+See the :ref:`shell <shell_api>` documentation for general instructions on how to connect and
+interact with the shell. The CAN shell comes with built-in help (unless
+:kconfig:option:`CONFIG_SHELL_HELP` is disabled). The built-in help messages can be printed by
+passing ``-h`` or ``--help`` to the ``can`` command or any of its subcommands. All subcommands also
+support tab-completion of their arguments.
+
+.. tip::
+   All of the CAN shell subcommands take the name to a CAN controller as their first argument. A
+   list of all devices available can be obtained using the ``device list`` shell command when
+   :kconfig:option:`CONFIG_DEVICE_SHELL` is enabled. The examples below all use the device name
+   ``can@0``.
```

Comment:
is tab completion for device name also supported (a quick look at the code makes me feel it is)? If yes, this could be a useful thing to mention as part of the tip?

Reply:
Good idea. I have added this to the tip.